app/forms/idv/doc_pii_form.rb

```diff
@@ -58,8 +58,29 @@ def self.pii_like_keypaths
       keypaths
     end
 
+    # Modifies the errors object, used in image_upload_response_presenter to customize
+    # error messages for rendering  pii errors
+    #
+    # errors: The DocPiiForm errors object
+    def self.present_error(existing_errors)
+      return if existing_errors.blank?
+      if existing_errors.any? do |k, v|
+        PII_ERROR_KEYS.include?(k)
+      end
+        existing_errors[:front] = [I18n.t('doc_auth.errors.general.multiple_front_id_failures')]
+        existing_errors[:back] = [I18n.t('doc_auth.errors.general.multiple_back_id_failures')]
+      end
+
+      if existing_errors.many? { |k, v| %i[name dob dob_min_age state].include?(k) }
```

Comment:
(I know this was moved over from `image_upload_response_presenter` and I am asking this question for my own learning process)

Why is it `name dob dob_min_age state` here and not `PII_ERROR_KEYS`?

Reply:
the business logic has been that if more than one of these fields has an error, a generic error is created ... and if other errors occur then we create an error for that specific error.

👀 https://github.com/18F/identity-idp/blob/4d0c4312a0bac39bfaf2f8bef9c170d67b89dfbb/app/forms/idv/doc_pii_form.rb#L85

Reply:
Thanks!